### ISSUE TITLE
Blank pages after upgrading to 4.3.1

### ISSUE THREAD

**Issue by lionpanthera** [mastodon/mastodon#32936]
After upgrading from 4.2.0 to 4.3.1 on Debian 11, Mastodon is now returning completely blank dark grey pages.

There were no obvious errors during the upgrade process so I have no idea what to do now.

Things I did:

- Upgraded Ruby
- Installed Yarn 4 with corepack
- Did bundle install and yarn install --immutable

This step produced the most questionable output, complaining about dependencies with overlapping version ranges and then finally saying "The lockfile would have been modified by this install, which is explicitly forbidden." Other bugs say that this is OK, so, maybe it's OK?

- Did rails db:encryption:init and put them in .env.production
- Did bundle exec rails assets:clobber
- Did bundle exec rails db:migrate with SKIP_POST_DEPLOYMENT_MIGRATIONS=true
- Did bundle exec rails assets:precompile
- Restarted services
- Did bundle exec rails db:migrate

And now my server is returning blank pages.

I've repeated the clobber/migration/precompile steps in a variety of orders but nothing has helped. Any suggestions?

**Comment by renchap** [comment]
As you say that yarn tells you that the lockfile would be updated, thats probably the issue.

I suspect you have a custom change in your `package.json`. Revert to the one from upstream, then run `yarn install --immutable` again,  remove `tmp/cache/webpacker` and recompile your assets.

**Comment by lionpanthera** [comment]
I don't believe I've made any changes to package.json, but I did checkout again and followed your advice.

yarn install --immutable stopped complaining about the lockfile but still complained about dependency versions.

precompile complained that some gems were missing, so I did bundle install again. precompile then worked, but threw a lot of warnings - but it always does, so I assume that's OK?

However my pages are still blank. No apparent change.
